src/core/index.ts

```diff
@@ -284,10 +291,36 @@ export class ZarrArray {
       return out;
     }
 
-    for (const proj of indexer.iter()) {
-      await this.chunkGetItem(proj.chunkCoords, proj.chunkSelection, out, proj.outSelection, indexer.dropAxes);
+    const concurrencyLimit = (
+      options === undefined
+      || options === null
+      || options.concurrencyLimit === undefined
+      || options.concurrencyLimit === null
+    ) ? 10 : options.concurrencyLimit;
+
+    const progressCallback = (
+      options === undefined
+      || options === null
+      || options.progressCallback === undefined
+      || options.progressCallback === null
+    ) ? null : options.progressCallback;
+
+    const limit = pLimit(concurrencyLimit);
+    const input = Array.from(
+      indexer.iter(),
```

Comment:
Random thought: I wonder if this has different memory implications, my guess is that the whole of the iterator is consumed and kept in memory. Which is fine, this should never really be huge (the sum of all projections)

Reply:
I thought about this as well, but I didn't have a good solution how to find the number of chunks otherwise, and `p-limit` (and other concurrency limiters I've seen) require an array. I can do some more digging tonight ...

Reply:
I'm happy to merge it as is just consuming the whole array. Its O(N) with the amount of chunks (and ~3 numbers for each), no need to optimize that to get this merged